### DISCUSSION TITLE
The magnitude of input GuassianBeam is not equals to 1

### DISCUSSION THREAD

**Discussion by beizhixing** [NanoComp/meep#2850] [Q&A] [answered]
I want to simulate a Circular-polarized Gaussianbeam input with mangitude is 1. However, the simulations of input plane in vacuum shows it is vear small, I have no idea, is there any details I need to set?

Here is the code and the resulting image:
   # Circular-polarized gaussian continous source
    beam_x0 = mp.Vector3(0, 0, 0)  # beam focus (relative to source center)
    rot_angle = 0  # CCW rotation angle about z axis (0: +y axis)
    beam_kdir = mp.Vector3(0, 0, 1).rotate(
        mp.Vector3(0, 1, 0), math.radians(rot_angle)
    )  # beam propagation direction
    beam_w0 = r  # beam waist radius
    beam_E0 = math.sqrt(2)/2*mp.Vector3(1,1j,0)
    sources = [
    mp.GaussianBeamSource(
        src=mp.ContinuousSource(fcen, fwidth=df, is_integrated=True),
        center=mp.Vector3(0,0, -0.5 * sz + dpml+ 0.5*dair),
        size=mp.Vector3(sides,sides,0),
        beam_x0=beam_x0,
        beam_kdir=beam_kdir,
        beam_w0=beam_w0,
        beam_E0=beam_E0, 
        )
    ]
    
    # Free-space
    sim = mp.Simulation(
        resolution=resolution,
        cell_size=cell_size,
        boundary_layers=pml_layers,
        sources=sources,
        default_material=mp.Medium(index=1.0)
    )
    
    xy_sizes = mp.Vector3(sides, sides, 0)
    sim.add_flux(fcen, 0, 1, mp.FluxRegion(center=mp.Vector3(0, 0, 0), size=mp.Vector3(sides, 0, d)))
    sim.add_flux(fcen, 0, 1, mp.FluxRegion(center=mp.Vector3(0, 0, -0.5 * sz + dpml + 0.5 * dair), size=xy_sizes))
    
    sim.run(until = 1000)

    # Obtain electric field data from the monitors
    plin_data = [sim.get_array(center=mp.Vector3(0, 0, -0.5 * sz + dpml + 0.5*dair), size=mp.Vector3(sides, sides, 0), component=c, cmplx=True) for c in [mp.Ex, mp.Ey, mp.Ez, mp.Hx, mp.Hy, mp.Hz]]    
    plXZ_data = [sim.get_array(center=mp.Vector3(0,0,0), size=mp.Vector3(sides, 0, d), component=c, cmplx=True) for c in [mp.Ex, mp.Ey, mp.Ez, mp.Hx, mp.Hy, mp.Hz]]
    
    plane_in_intensity = np.sqrt(np.abs(plin_data[0])**2 + np.abs(plin_data[1])**2 + np.abs(plin_data[2])**2)
    plane_XZ_intensity = np.sqrt(np.abs(plXZ_data[0])**2 + np.abs(plXZ_data[1])**2 + np.abs(plXZ_data[2])**2)
    savemat('Vacuum_in_data.mat', {'Ex': plin_data[0], 'Ey': plin_data[1], 'Ez': plin_data[2], 'Hx': plin_data[3], 'Hy': plin_data[4], 'Hz': plin_data[5]})
    savemat('Vacuum_XZ_data.mat', {'Ex': plXZ_data[0], 'Ey': plXZ_data[1], 'Ez': plXZ_data[2],'Hx': plXZ_data[3], 'Hy': plXZ_data[4], 'Hz': plXZ_data[5]})

<img src="https://github.com/NanoComp/meep/assets/27534638/3946f403-e78a-4283-bd9e-65dd17115e67" width="200" height="180" />

<img src="https://github.com/NanoComp/meep/assets/27534638/84dc1443-48e1-434f-b74d-4458f6db2628"  width="270" height="150" Align="Right" />

**Comment by stevengj** [comment]
From your second plot, it looks like |E| **at the focal point** is equal to 1.  Of course, it is much smaller elsewhere due to spatial dispersion.   If you want to reduce the spatial dispersion you need to increase the beam waist.

**Reply by beizhixing** [reply]
I know the condition you mentioned but the question is I can not understand why the |E| in the input plane(**XY plane**) at the focal point |Ein| not show maximum 1. (The right figure is Ey)

<img src="https://github.com/NanoComp/meep/assets/27534638/9e952219-4bf3-489f-ae13-6fd1a4c224d0" width="300" height="270" />

<img src="https://github.com/NanoComp/meep/assets/27534638/5fb7e12f-1efa-41bc-838f-3bc667ead8b0" width="210" height="300" />

**Comment by stevengj** [comment, marked as answer]
You have to be careful about slicing it exactly at the input source plane, because E is discontinuous across that plane. Due to Meep’s interpolation, you will get some kind of average of the zero fields below the plane with the large fields just above the plane. Try looking a pixel or two above the plane. 

And if you are looking at time-domain fields, you might want to set the force-complex-fields option to True so that you get the full complex amplitude and not just the real part of the field.

**Reply by beizhixing** [reply]
Under your suggestion, I set the monitor above the source 0.1μm, and I get a better result as follows，**_the maximum of |E| is 0.69_**.

<img src="https://github.com/NanoComp/meep/assets/27534638/2b6fb48b-62e7-4a21-b365-a5d3e9d24c40" width="300" height="270" />

 ####About the complex filed, I use the code like:
plin_data = [sim.get_array(center=mp.Vector3(0, 0, -0.5 * sz + dpml + 0.6*dair), size=mp.Vector3(sides, sides, 0), component=c, **cmplx=True**) for c in [mp.Ex, mp.Ey, mp.Ez, mp.Hx, mp.Hy, mp.Hz]]    

Is it correct?
